lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -16,7 +16,7 @@ def initialize(namespace:, context:, definition:, logger:, statsd_tags: nil,
 
     def sync(cache)
       super
-      @pods = exists? ? find_pods(cache) : []
+      @pods = check_pods? ? find_pods(cache) : []
```

Comment:
This is called repeatedly until the resource is "done" (succeeded/failed/timed out), and the data retrieved on the previous cycle is no longer valid in subsequent cycles. For example, maybe on the first cycle we see 2 exist / 0 available, so we fetch them, and the look fine, so we continue. On the next cycle, we see 4 exist / 2 available, so we now don't bother fetching them. If we kept the stale data, anything that didn't respect `check_pods?` and actually used the data would draw incorrect conclusions. `@pods == []` already has to be handled as "dunno, try again later" in case of fetch errors, but the possibility that data is present but stale does not.